torch/testing/_internal/common_nn.py

```diff
@@ -4555,15 +4612,16 @@ def fw(input):
             res += torch.cat([get_numerical_jacobian(fw, input, p, eps=1e-6) for p in param], 0),
         return res
 
-    def check_jacobian(self, module, input, jacobian_input=True):
+    def check_jacobian(self, module, input: _TensorOrTensors, jacobian_input=True):
         jacobian_parameters = bool(self._get_parameters(module)[0])
         analytical = self._analytical_jacobian(module, input, jacobian_input, jacobian_parameters)
         numerical = self._numerical_jacobian(module, input, jacobian_input, jacobian_parameters)
         analytical_t = list(iter_tensors(analytical))
         numerical_t = list(iter_tensors(numerical))
 
         # TODO: compare structure
-        if input.numel() != 0:
+        input_tuple = input if isinstance(input, tuple) else (input,)
+        if any(t.numel() != 0 for t in input_tuple):
```

Comment:
What was the initial `if input.numel()` check guarding against? I'd like to make sure it is safe to do `if any(t.numel() != 0 for t in input_tuple)` here

Reply:
This is not a guard, just to save program running time. Previously we have only one input,  if that input is empty, then the jacobian output size will be 0 (jacobian output size is equal to the forward output's numel ), then no need to compare numerical jacobian and analytical jacobian. 
Now we have multiple tensors, so if any of them is not empty, then I assume we need to compare jacobian.

Reply:
Let's say the input was an empty tensor (so the input has size [0]). Then `input.grad` would have size `[0]`. The analytical jacobian would also have 0 numel (I think it get size [0, 0], but that doesn't really matter).

So in this case, `analytical_t = [ torch.zeros([0, 0]) ]`, `numerical_t = [ torch.zeros([0, 0]) ]`. 
If we tried to do `a.sub(n).abs().max()`, calling `.max()` on a tensor with no numel throws an exception!  So I think what we're guarding against here is calling .max() on a tensor with 0 numel().

I'm not 100% sure that the numerical jacobian actually comes out to be an empty tensor (it really depends on how we compute it...). But if the above is correct, then we should change the check to something like:
```
differences = []
for input, a, n in zip(input_tuple, analytical_t, numerical_t):
    if a.numel() == 0:
        continue   # TODO: compare structure (ensure analytic jacobian has correct shape)
    differences.append(a.add(n, alpha=-1).abs().max())
if differences):
    self.assertLessEqual(max(differences), PRECISION)
```

Reply:
@zou3519 
Ah, I did not check whether max will throw an error on empty tensor. I looked into the jacobian code, as long as one of input is not empty, the output size from jacobian will not be empty based on our current module. You are right, if max throw, the code will need to be like what you said. I will update the PR